src/main/kotlin/assignment/Assignment.kt

```diff
@@ -21,13 +24,35 @@ fun main() {
 }
 
 fun readResponses(): List<Response> {
-    TODO()
+    var returnList = mutableListOf<Response>()
+    val sc = Scanner(File("responses.txt"))
+    sc.nextLine()
+
+    while (sc.hasNextLine()) {
+        sc.next()
+        returnList.add(Response(sc.next(), sc.nextInt(), sc.next(), sc.next() == "yes"));
+    }
+    sc.close()
+    returnList = returnList.filterNot { it.covidPositive }.toMutableList()
+    returnList.sortWith(compareBy<Response> {it.branch}.thenBy {it.name})
```

Comment:
This should be part of the `processResponses` function. `readResponses()` is just for reading the data only.